src/EditorFeatures/CSharpTest/Diagnostics/ConditionalExpressionInStringInterpolation/CSharpAddParenthesisAroundConditionalExpressionInInterpolatedStringCodeFixProviderTests.cs

```diff
@@ -0,0 +1,100 @@
+            => (null, new CSharpAddParenthesisAroundConditionalExpressionInInterpolatedStringCodeFixProvider());
+
+        private async Task TestInMethodAsync(string initialMethodBody, string expectedMethodBody)
+        {
+            var template = @"
+class Application
+{{
+    public static M()
+    {{
+        {0}
+    }}
+}}";
+            await TestInRegularAndScriptAsync(
+                string.Format(template, initialMethodBody), 
+                string.Format(template, expectedMethodBody)).ConfigureAwait(false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesisSimpleConditionalExpression()
+        {
+            await TestInMethodAsync(
+                @"var s = $""{ true ? 1 [|:|] 2}"";",
+                @"var s = $""{ (true ? 1 : 2)}"";");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesisMultiLineConditionalExpression1()
+        {
+            await TestInMethodAsync(@"
+var s = $@""{ true
+            [|? 1|]
+            : 2}"";
+", @"
+var s = $@""{ (true
+            ? 1
+            : 2)}"";
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesisMultiLineConditionalExpression2()
+        {
+            await TestInMethodAsync(@"
+var s = $@""{
+            true
+            ?
+            [|1|]
+            : 
+            2
+            }"";
+", @"
+var s = $@""{
+            (true
+            ?
+            1
+            : 
+            2)
+            }"";
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesisWithTrivia()
+        {
+            await TestInMethodAsync(
+                @"var s = $""{ /* Leading1 */ true /* Leading2 */ ? /* TruePart1 */ 1 /* TruePart2 */[|:|] /* FalsePart1 */ 2 /* FalsePart2 */ }"";",
+                @"var s = $""{ /* Leading1 */ (true /* Leading2 */ ? /* TruePart1 */ 1 /* TruePart2 */: /* FalsePart1 */ 2 /* FalsePart2 */ )}"";");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesis_ClosingBracketInFalseCondition()
+        {
+            await TestInMethodAsync(
+                @"var s = $""{ true ? new int[0] [|:|] new int[] {} }"";",
+                @"var s = $""{ (true ? new int[0] : new int[] {} )}"";");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParenthesis)]
+        public async Task TestAddParenthesis_StringLiteralInFalseCondition()
+        {
+            await TestInMethodAsync(
+                @"var s = $""{ true ? ""1"" [|:|] ""2"" }"";",
+                @"var s = $""{ (true ? ""1"" : ""2"" )}"";");
```

Comment:
Needs fix all test.

Reply:
I plan to add (lots) of tests next week (I rushed this out to get feedback for the nasty string problem described in the PR comments). But I don't think FixAll is useful for this CodeFixProvider.

Reply:
Fair enough :)